CONTRIBUTING.md

```diff
@@ -35,3 +36,79 @@ documentation.
 `just docs` : Generates Go docs and starts http server on localhost
 
 `just bootstrap` : Installs necessary go packages for gen and docs
+
+## Testing
+
+The `go-pinecone` codebase includes both unit and integration tests. These tests are kept within the same files, but are
+constructed differently. See `/pinecone/index_connection_test.go` and `/pinecone/client_test.go` for examples. They are divided into sections with `// Integration tests: ` near the top, and `// Unit tests:` near the bottom of the file.
+
+For running tests you can use `just test` to run all tests, and `just test-unit` to only run integration tests.
+
+### Unit tests
+
+Unit tests are generally written using Go's built-in support. You can find a [brief walkthrough](https://go.dev/doc/tutorial/add-a-test) detailing how to write a test. You can also refer to [go.dev/doc/code#Testing](https://go.dev/doc/code#Testing).
+
+When adding unit tests, make sure to postfix `"Unit"` to the test function name in order for the test to be picked up by the `just test-unit` command. For example:
+
+```Go
+func TestNewClientParamsSetUnit(t *testing.T) {
+	apiKey := "test-api-key"
+	client, err := NewClient(NewClientParams{ApiKey: apiKey})
+
+	require.NoError(t, err)
+	require.Empty(t, client.sourceTag, "Expected client to have empty sourceTag")
+	require.NotNil(t, client.headers, "Expected client headers to not be nil")
+	apiKeyHeader, ok := client.headers["Api-Key"]
+	require.True(t, ok, "Expected client to have an 'Api-Key' header")
+	require.Equal(t, apiKey, apiKeyHeader, "Expected 'Api-Key' header to match provided ApiKey")
+	require.Equal(t, 3, len(client.restClient.RequestEditors), "Expected client to have correct number of request editors")
+}
+```
+
+### Integration Tests
+
+For integration tests we use the `stretchr/testify` module, specifically for the `suite`, `assert`, and `require` packages. You can find the source code and documentation on GitHub: [https://github.com/stretchr/testify](https://github.com/stretchr/testify).
+
+There are two files that define the integration test suite, and include code that manages setup and teardown of external Index resources before the test files are executed:
```

Comment:
Nit: I'd remove "before the test files..." because technically the teardown happens after they're executed, right?